pkg/codegen/customresourcegenerator.go

```diff
@@ -16,18 +16,26 @@ package codegen
 
 import (
 	"fmt"
+	"strings"
 
-	"github.com/pulumi/crd2pulumi/internal/unstruct"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"github.com/go-openapi/jsonreference"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/cgstrings"
+	extensionv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+const (
+	definitionPrefix = "#/definitions/"
 )
 
 // CustomResourceGenerator generates a Pulumi schema for a single CustomResource
 type CustomResourceGenerator struct {
 	// CustomResourceDefinition contains the unmarshalled CRD YAML
-	CustomResourceDefinition unstructured.Unstructured
+	CustomResourceDefinition extensionv1.CustomResourceDefinition
 	// Schemas represents a mapping from each version in the `spec.versions`
 	// list to its corresponding `openAPIV3Schema` field in the CRD YAML
-	Schemas map[string]map[string]any
+	Schemas map[string]spec.Swagger
```

Comment:
Does k8s just happen to use older type names for back-compact? I guess it probably pre-dates the Swagger→OpenAPI change so it would make sense. I just want to double check we're using whatever the most recent/relevant logic is.

Reply:
Newer k8 versions provide both v2 and v3 specs, so it might be nice to upgrade to v3 specs in the future for p-k schemagen. Also should note that p-k merges schemas from a range of Kubernetes versions, so we may not want to do this in the short term unless we drop support for these older versions that do not serve a v3 spec.